management/tests/mocks/fetch-management.ts

```diff
@@ -0,0 +1,43 @@
+import { execute, makePromise, GraphQLRequest, FetchResult } from "apollo-link";
+import { setContext } from "apollo-link-context";
+import { HttpLink } from "apollo-link-http";
+import fetch from "cross-fetch";
+import { FetchError } from "node-fetch";
+
+import { ConnectUnreachableError } from "../../src/errors";
+import { ManagementCredentials } from "../../src/types";
+
+async function mockFetchManagement<T = unknown>(
+  managementCredentials: ManagementCredentials,
+  operation: GraphQLRequest,
+  headerValue: { behaviour: string; targetedFailure: string },
+): Promise<FetchResult<T>> {
+  const httpLink = new HttpLink({
+    uri: managementCredentials.URI,
+    fetch,
+  });
+
+  const authLink = setContext((_, { headers }) => {
+    return {
+      headers: {
+        ...headers,
+        authorization: `API_KEY ${managementCredentials.APIKey}`,
+        behaviour: headerValue.behaviour,
+        "targeted-failure": headerValue.targetedFailure,
```

Comment:
Rather than adding pretty much the same function that the initial one, what do you think of mocking `execute` instead?
It could still be possible to either return the body of the response, either throw an error.

The advantage being, if we need to change something in the initial function, we wouldn't need to remind ourselves to change the code here directly: if we change code in `fetchManagement` without changing it here, that means some code don't use the real implementation of our code and success in tests but fail in another context